Bottom: 7247cea880068c17944afcc149d4b6a8c73649ac
Top:    0f7d0508d7f04f2c277a3243cff503a97862859a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-27 02:12:27 +0100

hid/gtk: g_return_if_reached() for receipt of gui->use_mask(HID_MASK_BEFORE)

This means one less untranslated printf which is never hit. If for some
reason the core sends us one of these - logging a warning and continuing
is probably better than a hard abort()


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 52d9990..76e7659 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -270,8 +270,8 @@ ghid_use_mask (int use_it)
       break;
 
     case HID_MASK_BEFORE:
-      printf ("gtk doesn't support mask_before!\n");
-      abort ();
+      /* The HID asks not to receive this mask type, so warn if we get it */
+      g_return_if_reached ();
 
     case HID_MASK_CLEAR:
       if (!gport->mask)
